docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,386 @@
 # Writing ngraph transformations {#new_ngraph_transformation}
 
-1. Code of such transformation MUST be directly in template plugin soruces.
-2. It must be mark with doxygen markeds like
+This guide contains all necessary information that could help you to start writing nGraph transformations.
 
+First of all before writing transformation make sure that there is no transformation with the same functionality in [transformation library].
+To start writing transformation it's good to know how [transformation library] is structured, how transformations are organized and where to put your transformation code.
 
-// ! [new_transformation:part1]
+Let's start from reviewing transformations library structure.
+Transformations library is independent from InferenceEngine target library and located in `inference-engine/src/transformations` directory.
+Transformations root directory contains two folders:
+1. ngraph_ops - legacy opset operations needed for nGraph to CNNNetwork conversion.
```

Comment:
do we need to put this information about legacy opset? Maybe it's a good to split transformation library into 2 parts and legacy part can be moved to `inference_engine_legacy`

Reply:
I added notes about restricted usage of this legacy things in new plugins.